packages/client/src/encryption/RSAKeyPair.ts

```diff
@@ -41,59 +40,31 @@ async function exportCryptoKey(key: CryptoKey, { isPrivate = false } = {}): Prom
     return `-----BEGIN ${TYPE} KEY-----\n${exportedAsBase64}\n-----END ${TYPE} KEY-----\n`
 }
 
-// after RSAKeyPair is ready
-type InitializedRSAKeyPair = O.Overwrite<RSAKeyPair, {
-    privateKey: string
-    publicKey: string
-}>
-
 export class RSAKeyPair {
-    /**
-     * Creates a new instance + waits for ready.
-     * Convenience.
-     */
-    static async create(): Promise<RSAKeyPair> {
-        const pair = new RSAKeyPair()
-        await pair.onReady()
-        return pair
-    }
-
-    public privateKey: string | undefined
-    public publicKey: string | undefined
-    private _generateKeyPairPromise: Promise<void> | undefined
-
-    async onReady(): Promise<void> {
-        if (this.isReady()) { return undefined }
-        return this.generateKeyPair()
-    }
+    // the keys are in PEM format
+    public privateKey: string
+    public publicKey: string
 
-    isReady(this: RSAKeyPair): this is InitializedRSAKeyPair {
-        return (this.privateKey !== undefined && this.publicKey !== undefined)
+    private constructor(privateKey: string, publicKey: string) {
+        this.privateKey = privateKey
+        this.publicKey = publicKey
     }
 
-    // Returns a String (base64 encoding)
     getPublicKey(): string {
-        if (!this.isReady()) { throw new Error('RSAKeyPair not ready.') }
         return this.publicKey
     }
 
-    // Returns a String (base64 encoding)
     getPrivateKey(): string {
-        if (!this.isReady()) { throw new Error('RSAKeyPair not ready.') }
         return this.privateKey
     }
 
-    /* eslint-disable no-underscore-dangle */
-    private async generateKeyPair(): Promise<void> {
-        if (!this._generateKeyPairPromise) {
-            this._generateKeyPairPromise = (typeof window !== 'undefined')
-                ? this.keyPairBrowser()
-                : this.keyPairServer()
-        }
-        return this._generateKeyPairPromise
+    static async create(): Promise<RSAKeyPair> {
+        return (typeof window !== 'undefined')
+            ? RSAKeyPair.create_browserEnvironment()
+            : RSAKeyPair.create_serverEnvironment()
```

Comment:
Why not typical naming convention?
```
RSAKeyPair.createBrowserEnvironment()
RSAKeyPair.createServerEnvironment()
```

Reply:
Not sure if we have a convention for methods? Usually we have a separate class, e.g. `BrowserPersistence` and `ServerPersistence`.

Without underscores the methods would would create _environments_, not _key pairs_. E.g. the earlier name `keyPairBrowser` was very confusing: it sounded like it'd return some kind of searching iterator (a method which _browses_ a key from a set of keys).